src/analytics/lambdas/user-segments-workflow/segment-job-status.ts

```diff
@@ -64,20 +74,95 @@ const _handler = async (event: SegmentJobStatusEvent) => {
         ReturnValues: 'ALL_NEW',
       });
 
-      // TODO: retrieve segment result including sample data from S3, update result to job status table
-
       logger.info(`Update segment job status to ${jobStatus} in DDB metadata table`);
       await ddbDocClient.send(command);
+    } else if (jobStatus === SegmentJobStatus.COMPLETED) {
+      const s3Path = `${process.env.SEGMENTS_S3_PREFIX}app/${appId}/segment/${segmentId}/job/${jobRunId}/`;
+
+      // Retrieve segment result including sample data from S3, update job status table
+      const summaryData = await s3Client.send(new GetObjectCommand({
+        Bucket: process.env.PIPELINE_S3_BUCKET,
+        Key: `${s3Path}segment-summary_000.csv`,
+      }));
+      if (summaryData.Body) {
+        // Parse segment summary file
+        const content = await summaryData.Body.transformToString();
+        const lines = content.split('\n');
+        const values = lines[1].split(',');
+        const segmentUserNumber = values[0];
+        const totalUserNumber = values[1];
+        const endTime = values[2];
+
+        // Get sample data from segment output
+        const sampleData = await readSegmentSampleDataFromS3(process.env.PIPELINE_S3_BUCKET!, `${s3Path}segment_000.csv`);
+
+        // Update segment job status in DDB
+        const command = new UpdateCommand({
+          TableName: ddbTableName,
+          Key: {
+            id: event.segmentId,
+            type: `SEGMENT_JOB#${event.jobRunId}`,
+          },
+          UpdateExpression: 'set jobStatus = :js, jobEndTime = :et, segmentUserNumber = :su, totalUserNumber = :tu, sampleData = :sd',
+          ExpressionAttributeValues: {
+            ':js': jobStatus,
+            ':et': endTime,
+            ':su': segmentUserNumber,
+            ':tu': totalUserNumber,
+            ':sd': sampleData,
+          },
+          ReturnValues: 'ALL_NEW',
+        });
+
+        logger.info(`Update segment job status to ${jobStatus} in DDB metadata table`);
+        await ddbDocClient.send(command);
+      }
     }
 
     return {
       ...event,
       jobStatus,
     };
   } catch (err) {
-    logger.error('Error when executing segment query.', err as Error);
+    logger.error('Error when checking segment job status.', err as Error);
     throw err;
   }
 };
 
+async function readSegmentSampleDataFromS3(bucketName: string, key: string): Promise<any[]> {
```

Comment:
How about using `readS3ObjectAsString` from `common/s3`?

Reply:
That helper function will read the whole file. Since the segment output file size could be pretty large, we'd better only load the file partially to get sample data.

In this function, it read up to first 50 lines from S3 file stream. Also it leverages `csv-parser` library to parse the csv data.